### PR TITLE
Vmcommon to core

### PR DESCRIPTION
**Pull request by sasurobert**
Moved vm-common to core. removed the dependencies to the old repository.

### CODE REVIEW

**Review by camilbancioiu — DISMISSED** [review]

File: core/vm-common/codeMetadata.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package vmcommon
+
+const lengthOfCodeMetadata = 2
+
+const (
+	METADATA_UPGRADEABLE = 1
```
Comment (iulianpascalau) [inline core/vm-common/codeMetadata.go:6]: Please rename these to match with the rest of the project: MetadataUpgradeable....

File: core/vm-common/executionHandler.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+// VMExecutionHandler interface for any Elrond VM endpoint
+type VMExecutionHandler interface {
+	// Computes how a smart contract creation should be performed
+	RunSmartContractCreate(input *ContractCreateInput) (*VMOutput, error)
+
+	// Computes the result of a smart contract call and how the system must change after the execution
+	RunSmartContractCall(input *ContractCallInput) (*VMOutput, error)
+
+	// GasScheduleChange sets a new gas schedule for the VM
+	GasScheduleChange(newGasSchedule map[string]map[string]uint64)
```
Comment (iulianpascalau) [inline core/vm-common/executionHandler.go:12]: Please add IsInterfaceNil here

File: core/vm-common/hookBlockchain.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	IsPayable(address []byte) (bool, error)
+
+	// SaveCompiledCode saves to cache and storage the compiled code
+	SaveCompiledCode(codeHash []byte, code []byte)
+
+	// GetCompiledCode returns the compiled code if it finds in the cache or storage
+	GetCompiledCode(codeHash []byte) (bool, []byte)
+
+	// ClearCompiledCodes clears the cache and storage of compiled codes
+	ClearCompiledCodes()
```
Comment (iulianpascalau) [inline core/vm-common/hookBlockchain.go:86]: Please add IsInterfaceNil

File: core/vm-common/hookBlockchain.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
+package vmcommon
```
Comment (iulianpascalau) [inline core/vm-common/hookBlockchain.go:1]: this should be renamed to interface.go

File: core/vm-common/executionHandler.go [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package vmcommon
+
+// VMExecutionHandler interface for any Elrond VM endpoint
+type VMExecutionHandler interface {
```
Comment (iulianpascalau) [inline core/vm-common/executionHandler.go:4]: please move in the next file (interface.go)

File: core/vm-common/hookCrypto.go [hunk 6]
```diff
@@ -0,0 +1,17 @@
+package vmcommon
+
+// CryptoHook interface for VM krypto functions
+type CryptoHook interface {
```
Comment (iulianpascalau) [inline core/vm-common/hookCrypto.go:4]: Please move to interface.go and add IsInterfaceNil method

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

File: core/vm-common/codeMetadata.go (hunk 1 above)
Comment (sasurobert) [reply core/vm-common/codeMetadata.go:6]: it was copy paste from vmcommon. not necessarily to update everywhere.

Comment (sasurobert) [reply core/vm-common/codeMetadata.go:6]: done

File: core/vm-common/hookBlockchain.go (hunk 3 above)
Comment (sasurobert) [reply core/vm-common/hookBlockchain.go:86]: done

File: core/vm-common/hookBlockchain.go (hunk 4 above)
Comment (sasurobert) [reply core/vm-common/hookBlockchain.go:1]: done

File: core/vm-common/executionHandler.go (hunk 5 above)
Comment (sasurobert) [reply core/vm-common/executionHandler.go:4]: done

File: core/vm-common/hookCrypto.go (hunk 6 above)
Comment (sasurobert) [reply core/vm-common/hookCrypto.go:4]: done

**Review by iulianpascalau — DISMISSED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by camilbancioiu — APPROVED** [review]

**Pull request merged**